.github/workflows/publish_plugin.yml

```diff
@@ -153,4 +88,15 @@ jobs:
       - name: Publish snforge_scarb_plugin
         if: needs.check-version.outputs.plugin_uploaded == 'false' || github.event_name == 'workflow_dispatch'
         working-directory: crates/snforge-scarb-plugin
-        run: scarb publish --allow-dirty
+        run: |
+          set -exo pipefail
+          
+          if [ -n "${{ inputs.overridden_plugin_version }}" ]; then
+            sed -i.bak "/\[package\]/,/version =/ s/version = \".*/version = \"${{ inputs.overridden_plugin_version }}\"/" Scarb.toml
+            rm Scarb.toml.bak 2> /dev/null
+
+            sed -i.bak "/\[package\]/,/version =/ s/version = \".*/version = \"${{ inputs.overridden_plugin_version }}\"/" Cargo.toml
+            rm Cargo.toml.bak 2> /dev/null
+          fi
+
+          scarb publish --allow-dirty ${{ inputs.dev_registry == true && '--index https://scarbs.dev/' || '' }}
```

Comment:
I'm a bit afraid of this, mainly if it won't be too easy to upload a new/test version to prod by mistake. Maybe we should set dev registry to be a default instead?